docs/zetacore/gas_and_fees.md

```diff
@@ -0,0 +1,334 @@
+
+In both scenarios, the fees for the revert transaction are covered by the protocol. This means the user receives the full withdraw amount back.
+
+For `onRevert` hook calls, a fixed gas limit is used by the protocol.
+This gas limit can be retrieved with the [following query](https://zetachain.blockpi.network/lcd/v1/public/zeta-chain/fungible/system_contract) under the `gateway_gas_limit` field.
+
+### Aborting Withdraw
+
+Aborts occur when a revert transaction itself reverts or cannot be executed.
+
+When the user specifies an abort address, the protocol will attempt to call the `onAbort` hook on this address on ZetaChain.
+
+As with revert transactions, the fees for the abort transaction are covered by the protocol and use the fixed gas limit defined in the system contract.
+
+## Connected Chains Deposit Fees
+
+When making deposits from a connected chain to ZetaChain,
+users typically pay the inherent fees of the connected chain for the deposit transaction.
+The deposit call on ZetaChain is typically covered by the protocol.
+There are some exceptions per chain covered below.
+
+For deposits on ZetaChain that include a smart contract call, a fixed gas limit is used by the protocol for the call.
+This gas limit can be retrieved with the [following query](https://zetachain.blockpi.network/lcd/v1/public/zeta-chain/fungible/system_contract) under the `gateway_gas_limit` field.
+
+### EVM
+
+[Ethereum Gas Documentation](https://ethereum.org/developers/docs/gas/).
+
+When executing a transaction that includes a single deposit to ZetaChain, the user only pays the inherent fees of the connected EVM chain for the deposit transaction.
+
+As a protection against spam deposits, if the transaction contains multiple deposits to ZetaChain, an additional fee per deposit is charged.
+
+The additional fee is the `additionalActionFeeWei` parameter defined in the connected chain gateway contract.
+
+Therefore, the total additional fee for multiple deposits is calculated in wei (or other base unit) as:
+```
+Total Additional Fee = (Number of Deposits - 1) × additionalActionFeeWei
+```
+
+> **Note**: As of this writing, the multiple deposits mechanism is not yet implemented on ZetaChain mainnet. Subsequent deposits in a single transaction are currently ignored.
+
+### Bitcoin
+
+For Bitcoin transactions, refer to the [Bitcoin Fee Documentation](https://developer.bitcoin.org/devguide/transactions.html).
+
+In addition to the Bitcoin network fees, a depositor fee is deducted from the deposited amount.
+The depositor fee is charged to cover the cost of spending the deposited UTXO in the future.
+
+The logic to calculate this depositor fee can be found in the [following page](https://www.zetachain.com/docs/developers/chains/bitcoin#fees).
+
+### Solana
+
+[Solana Fee Documentation](https://solana.com/docs/core/fees).
+
+No additional fees are charged for deposits from the Solana network to ZetaChain.
```

Comment:
there is additional deposit fee hardcoded for solana, i think other chains might have something similar, there was discussion about ton recently